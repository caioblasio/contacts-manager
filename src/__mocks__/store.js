const contactsState = require("providers/Contacts/reducer").initialState;

const history = require("history").createBrowserHistory();
const configureStore = require("store").default;

const rootState = {
  contacts: contactsState
};

const store = configureStore(rootState);

module.exports = store;
