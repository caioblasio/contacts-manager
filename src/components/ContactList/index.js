import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import Divider from "@material-ui/core/Divider";
import Typography from "@material-ui/core/Typography";
import Avatar from "@material-ui/core/Avatar";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import CardActionArea from "@material-ui/core/CardActionArea";

import useStyles from "./styles";

const ContactsList = ({ data, shortView, isScrollable }) => {
  const classes = useStyles();

  return (
    <section {...(isScrollable ? { className: classes.root } : {})}>
      {data.map((contact, index) => (
        <React.Fragment key={contact.id}>
          <Card>
            <CardActionArea component={Link} to={`/contact/${contact.id}`}>
              <CardContent className={classes.card}>
                <div className={classes.avatar}>
                  <Avatar>{contact.name.split("")[0].toUpperCase()}</Avatar>
                </div>
                <div className={classes.content}>
                  <Typography
                    variant="body1"
                    component="p"
                    className={classes.flexBig}
                  >
                    {contact.name}
                  </Typography>
                  {!shortView && (
                    <>
                      <Typography
                        variant="body1"
                        component="p"
                        className={classes.flexBig}
                      >
                        {contact.email}
                      </Typography>
                      <Typography
                        variant="body1"
                        component="p"
                        className={classes.flexBig}
                      >
                        {contact.phone}
                      </Typography>
                    </>
                  )}
                </div>
              </CardContent>
            </CardActionArea>
          </Card>
          {index !== data.length - 1 && <Divider />}
        </React.Fragment>
      ))}
    </section>
  );
};

ContactsList.defaultProps = {
  shortView: false,
  isScrollable: false
};

ContactsList.propTypes = {
  data: PropTypes.array,
  shortView: PropTypes.bool,
  isScrollable: PropTypes.bool
};

export default ContactsList;
