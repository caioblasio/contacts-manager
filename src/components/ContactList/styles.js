import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles((theme) => ({
  root: {
    paddingBottom: theme.spacing(8),
  },
  scrollable: {
    height: "100vh",
    overflow: "scroll",
    boxShadow:
      "0px 2px 1px -1px rgba(0,0,0,0.2), 0px 1px 1px 0px rgba(0,0,0,0.14), 0px 1px 3px 0px rgba(0,0,0,0.12)",
  },
  card: {
    display: "flex",
  },
  avatar: { marginRight: theme.spacing(4) },
  content: {
    flex: "1 0 0",
    [theme.breakpoints.up("sm")]: {
      display: "flex",
      alignItems: "center",
      "& > *": {
        flex: "1 0 0",
      },
    },
  },
}));

export default useStyles;
