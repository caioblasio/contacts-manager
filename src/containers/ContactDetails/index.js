import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { getContact } from "providers/Contacts/selectors";
import ContactDetails from "components/ContactDetails";

export const Container = ({ contact }) => {
  return contact ? (
    <ContactDetails contact={contact} />
  ) : (
    "Contact not found! :("
  );
};

const mapStateToProps = (_state, { id }) => {
  return createStructuredSelector({
    contact: getContact(id),
  });
};

Container.propTypes = {
  contact: PropTypes.object,
};

export default connect(mapStateToProps)(Container);
