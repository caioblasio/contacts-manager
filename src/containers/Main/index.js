import React, { useEffect } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Route, Switch } from "react-router-dom";
import CssBaseline from "@material-ui/core/CssBaseline";
import Home from "pages/Home";
import Details from "pages/Details";
import New from "pages/New";
import Topbar from "components/Topbar";
import { fetchContacts } from "providers/Contacts/actions";

export const Container = ({ fetchContacts }) => {
  useEffect(() => {
    fetchContacts();
  });
  return (
    <main className="main">
      <CssBaseline />
      <Topbar />
      <Switch>
        <Route exact path="/" component={Home} />
        <Route
          exact
          path="/contact/:id"
          render={props => <Details contactId={props.match.params.id} />}
        />
        <Route exact path="/new" component={New} />
      </Switch>
    </main>
  );
};

function mapDispatchToProps(dispatch) {
  return {
    fetchContacts: () => dispatch(fetchContacts())
  };
}

Container.propTypes = {
  fetchContacts: PropTypes.func.isRequired
};

export default connect(null, mapDispatchToProps)(Container);
