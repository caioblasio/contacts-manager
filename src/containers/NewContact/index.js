import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import NewContact from "components/NewContact";
import { addContact } from "providers/Contacts/actions";
import { push } from "providers/Router/actions";

export const Container = props => {
  return <NewContact {...props} />;
};

const mapDispatchToProps = dispatch => ({
  addContact: contact => dispatch(addContact(contact)),
  routePush: url => dispatch(push(url))
});

Container.propTypes = {
  addContact: PropTypes.func.isRequired,
  routePush: PropTypes.func.isRequired
};

export default connect(null, mapDispatchToProps)(Container);
