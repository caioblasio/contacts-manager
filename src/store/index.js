import { createStore, applyMiddleware, compose } from "redux";
import { createBrowserHistory } from "history";
import { routerMiddleware } from "connected-react-router/immutable";
import createSagaMiddleware from "redux-saga";
import createRootReducer from "./reducers";
import rootSaga from "./sagas";

export const history = createBrowserHistory();

export default function configureStore(initialState = {}) {
  let composeEnhancers = compose;
  const reduxSagaMonitorOptions = {};
  if (ENVIRONMENT !== "production" && typeof window === "object") {
    if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) {
      composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        name: "Redux",
        realtime: true,
        trace: true,
        traceLimit: 25
      });
    }
  }

  const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions);
  const middlewares = [sagaMiddleware, routerMiddleware(history)];
  const enhancers = [applyMiddleware(...middlewares)];

  const store = createStore(
    createRootReducer(history),
    initialState,
    composeEnhancers(...enhancers)
  );

  sagaMiddleware.run(rootSaga);
  return store;
}
