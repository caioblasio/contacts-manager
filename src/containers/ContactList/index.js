import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { getData } from "providers/Contacts/selectors";
import ContactList from "components/ContactList";

export const Container = props => {
  return <ContactList {...props} />;
};

const mapStateToProps = createStructuredSelector({
  data: getData()
});

Container.propTypes = {
  data: PropTypes.array
};

export default connect(mapStateToProps)(Container);
