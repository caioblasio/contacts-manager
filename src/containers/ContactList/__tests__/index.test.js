import React from "react";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import { Container as ContactListContainer } from "containers/ContactList";

import store from "__mocks__/store";
import { data } from "__mocks__/api";

describe("Given <ContactListContainer />", () => {
  let defaultProps;
  let component;
  let Container;

  defaultProps = {
    data
  };

  beforeAll(() => {
    Container = props => (
      <Provider store={store}>
        <MemoryRouter>
          <ContactListContainer {...defaultProps} {...props} />
        </MemoryRouter>
      </Provider>
    );
  });

  describe("When it initializes", () => {
    beforeEach(() => {
      const { container } = render(<Container />);
      component = container;
    });

    it("Then should match the snapshot", () => {
      expect(component).toMatchSnapshot();
    });
  });
});
