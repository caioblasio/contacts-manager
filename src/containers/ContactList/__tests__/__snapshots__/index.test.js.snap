// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <ContactListContainer /> When it initializes Then should match the snapshot 1`] = `
<div>
  <section>
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiCardActionArea-root"
        href="/contact/1"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiCardContent-root makeStyles-card-2"
        >
          <div
            class="makeStyles-avatar-3"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
            >
              L
            </div>
          </div>
          <div
            class="makeStyles-content-4"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Leanne Graham
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Sincere@april.biz
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              1-770-736-8031 x56442
            </p>
          </div>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiCardActionArea-root"
        href="/contact/2"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiCardContent-root makeStyles-card-2"
        >
          <div
            class="makeStyles-avatar-3"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
            >
              E
            </div>
          </div>
          <div
            class="makeStyles-content-4"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Ervin Howell
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Shanna@melissa.tv
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              010-692-6593 x09125
            </p>
          </div>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiCardActionArea-root"
        href="/contact/3"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiCardContent-root makeStyles-card-2"
        >
          <div
            class="makeStyles-avatar-3"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
            >
              C
            </div>
          </div>
          <div
            class="makeStyles-content-4"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Clementine Bauch
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Nathan@yesenia.net
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              1-463-123-4447
            </p>
          </div>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiCardActionArea-root"
        href="/contact/4"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiCardContent-root makeStyles-card-2"
        >
          <div
            class="makeStyles-avatar-3"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
            >
              P
            </div>
          </div>
          <div
            class="makeStyles-content-4"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Patricia Lebsack
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Julianne.OConner@kory.org
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              493-170-9623 x156
            </p>
          </div>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiCardActionArea-root"
        href="/contact/5"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiCardContent-root makeStyles-card-2"
        >
          <div
            class="makeStyles-avatar-3"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
            >
              C
            </div>
          </div>
          <div
            class="makeStyles-content-4"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Chelsey Dietrich
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Lucio_Hettinger@annie.ca
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              (254)954-1289
            </p>
          </div>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiCardActionArea-root"
        href="/contact/6"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiCardContent-root makeStyles-card-2"
        >
          <div
            class="makeStyles-avatar-3"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
            >
              M
            </div>
          </div>
          <div
            class="makeStyles-content-4"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Mrs. Dennis Schulist
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Karley_Dach@jasper.info
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              1-477-935-8478 x6430
            </p>
          </div>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiCardActionArea-root"
        href="/contact/7"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiCardContent-root makeStyles-card-2"
        >
          <div
            class="makeStyles-avatar-3"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
            >
              K
            </div>
          </div>
          <div
            class="makeStyles-content-4"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Kurtis Weissnat
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Telly.Hoeger@billy.biz
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              210.067.6132
            </p>
          </div>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiCardActionArea-root"
        href="/contact/8"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiCardContent-root makeStyles-card-2"
        >
          <div
            class="makeStyles-avatar-3"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
            >
              N
            </div>
          </div>
          <div
            class="makeStyles-content-4"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Nicholas Runolfsdottir V
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Sherwood@rosamond.me
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              586.493.6943 x140
            </p>
          </div>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiCardActionArea-root"
        href="/contact/9"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiCardContent-root makeStyles-card-2"
        >
          <div
            class="makeStyles-avatar-3"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
            >
              G
            </div>
          </div>
          <div
            class="makeStyles-content-4"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Glenna Reichert
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Chaim_McDermott@dana.io
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              (775)976-6794 x41206
            </p>
          </div>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiCardActionArea-root"
        href="/contact/10"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiCardContent-root makeStyles-card-2"
        >
          <div
            class="makeStyles-avatar-3"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
            >
              C
            </div>
          </div>
          <div
            class="makeStyles-content-4"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Clementina DuBuque
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Rey.Padberg@karina.biz
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              024-648-3804
            </p>
          </div>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
  </section>
</div>
`;
